Centaurus.Domain/PersistenceManager/PersistenceManager.cs

```diff
@@ -306,7 +295,7 @@ public async Task<Snapshot> GetSnapshot(long apex)
         public async Task<long> GetMinRevertApex()
         {
             //obtain min apex we can revert to
-            var minApex = await storage.GetFirstEffectApex();
+            var minApex = (await storage.LoadQuantaAboveApex(0, 1)).FirstOrDefault()?.Apex ?? -1;
```

Comment:
That looks scary. How about extracting a separate method for this? Why did you remove `GetFirstEffectApex` method?

Reply:
We don't have effects collection anymore, and we need to load first quantum to get the effect. But this looks really ugly. I'll create GetFirstEffectApex method.